static/js/statistics.ts

```diff
@@ -491,3 +493,145 @@ function updateChart(elementId: string, datasets: any[] ) {
   ch.data.datasets = datasets;
   ch.update();
 }
+
+export function resolve_student(class_id: string, error_id: string, prompt: string) {
+  modal.confirm(prompt, function(){
+    $.ajax({
+      type: 'DELETE',
+      url: '/live_stats/class/' + class_id + '/error/' + error_id,
+      contentType: 'application/json',
+      dataType: 'json'
+    }).done(function(response) {
+      if (response.achievement) {
+          showAchievements(response.achievement, true, "");
+      } else {
+          location.reload();
+      }
+    }).fail(function(err) {
+        modal.notifyError(err.responseText);
+    });
+  });
+}
+
+export function InitLineChart(data: any[], labels: any[]){
+  const ctx = document.getElementById("runsOverTime") as HTMLCanvasElement;
+  new Chart(ctx, {
+    type: 'line',
+    data: {
+      labels: labels.map(String),
+      datasets: [{
+        data: data,
+        fill: false,
+        pointBackgroundColor: function(context) {
+                    var index = context.dataIndex;
+                    var value = context.dataset.data[index];
+                    if (value === 0) {
+                    return 'red'
+                    }
+                    else if (value === 1){
+                      return 'green'
+                    }
+                    return 'blue'
+                },
+        // backgroundColor: 'rgba(0, 0, 255, 1)',
+        borderColor: 'rgba(0, 0, 255, 0.6)',
+        borderWidth: 1
+      }]
+    },
+    options: {
+      scales: {
+            y: {
+              ticks: {
+                callback: function(index) {
+                  // Hide every 2nd tick label
+                  if (index === 0) {
+                    return 'Fail'
+                  }
+                  else if (index === 1){
+                    return 'Success'
+                  }
+                  return ''
+                },
+              }
+            },
+        },
+      plugins: {
+        legend: {
+            display: false
+        }
+      }
+    }
+  });
+}
+
+export function enable_level_class_overview(level: string) {
```

Comment:
This file is dedicated for functions related to statistics and chart updates. This functions relates to the control of the page in a teacher's page, so its natural place is in the file `teachers.ts`

Reply:
Another thing I noticed is that these buttons require a refresh in order to select the levels that I want. I think is better to do this without the need to press a button to refressh, you can trigger a partial refresh using HTMX (the preferred method to handle interaction in Hedy currently) or use JavaScript and show and hide elements.